include/envoy/ssl/context_config.h

```diff
@@ -95,6 +96,19 @@ class ContextConfig {
    * @return The maximum TLS protocol version to negotiate.
    */
   virtual unsigned maxProtocolVersion() const PURE;
+
+  /**
+   * @return true if the ContextConfig is able to provide secrets to create SSL context,
+   * and false if dynamic secrets are expected but are not downloaded from SDS server yet.
+   */
+  virtual bool isReady() const PURE;
```

Comment:
Why do we need an explicit `isReady()?` Can't we just set the callback below and take an immediate invocation if ready?

Reply:
isReady() is used only when SDS sever sends bad response (i.e. no good secrets are received since Envoy starts), and the listenerImpl/ClusterImpl object has been marked as active state. We use isReady() to tell the context config is not ready.